src/psyclone/domain/lfric/lfric_extract_driver_creator.py

```diff
@@ -0,0 +1,974 @@
+        - variable that is read and written, it will create code to read in the
+          variable that is read, and create a new variable with the same name
+          and "_post" added which is read in to store the values from the
+          NetCDF file after the instrumented region was executed. In the end,
+          the variable that was read and written should have the same value
+          as the corresponding "_post" variable.
+        - variable that is written only, it will create a variable with "_post"
+          as postfix that reads in the output data from the NetCDF file. It
+          then also declares a variable without postfix (which will be the
+          parameter to the function), allocates it based on the shape of
+          the corresponding "_post" variable, and initialises it with 0.
+
+        :param program: the PSyIR Routine to which any code must \
+            be added. It also contains the symbol table to be used.
+        :type program: :py:class:`psyclone.psyir.nodes.Routine`
+        :param psy_data: the PSyData symbol to be used.
+        :type psy_data: :py:class:`psyclone.psyir.symbols.DataSymbol`
+        :param input_list: list of all signatures that are input variables \
+            to the instrumented region.
+        :type input_list: list of :py:class:`psyclone.core.Signature`
+        :param output_list: list of all signatures that are output \
+            variables of the instrumented region.
+        :type output_list: list of :py:class:`psyclone.core.Signature`
+        :param str postfix: a postfix that is added to a variable to \
+            create the corresponding variable that stores the output \
+            value from the kernel data file.
+
+        :returns: a list with all output parameters, i.e. variables that \
+            need to be verified after executing the kernel. Each entry is \
+            a 2-tuple containing the symbol of the computed variable, and \
+            the symbol of the variable that contains the value read from \
+            the file.
+        :rtype: list of 2-tuples of \
+            :py:class:`psyclone.psyir.symbols.Symbol`
+
+        '''
+        # pylint: disable=too-many-locals
+        all_sigs = list(set(input_list).union(set(output_list)))
+        all_sigs.sort()
+        symbol_table = program.scope.symbol_table
+        read_var = f"{psy_data.name}%ReadVariable"
+
+        # First handle variables that are read:
+        # -------------------------------------
+        for signature in input_list:
+            # Find the right symbol for the variable. Note that all variables
+            # in the input and output list have been detected as being used
+            # when the variable accesses were analysed. Therefore, these
+            # variables have References, and will already have been declared
+            # in the symbol table (in add_all_kernel_symbols).
+            sig_str = LFRicExtractDriverCreator.flatten_string(str(signature))
```

Comment:
Done. I like using the class to make it clear that this is a static function (which you otherwise can't see unless you look at the function definition). But I've fixed them all.